src/xcode/ENA/ENA/Source/HTTPClient/Resources/DCCRevocation/KIDTypeChunk/KIDTypeChunkResource.swift

```diff
@@ -0,0 +1,77 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+enum KIDTypeChunkResourceError: Error {
+	case DCC_RL_KTXY_CHUNK_SERVER_ERROR
+	case DCC_RL_KTXY_CHUNK_CLIENT_ERROR
+	case DCC_RL_KTXY_CHUNK_NO_NETWORK
+	case DCC_RL_KTXY_INVALID_SIGNATURE
+}
+
+struct KIDTypeChunkResource: Resource {
+
+	init(
+		kid: String,
+		hashType: String,
+		x: String,
+		y: String,
+		signatureVerifier: SignatureVerification = SignatureVerifier()
+	) {
+		self.trustEvaluation = DefaultTrustEvaluation(
+			publicKeyHash: Environments().currentEnvironment().pinningKeyHashData
+		)
+		self.locator = Locator(
+			kid: kid,
+			hashType: hashType,
+			x: x,
+			y: y
+		)
+		self.type = .caching()
+		self.sendResource = EmptySendResource()
+		self.receiveResource = ProtobufReceiveResource<SAP_Internal_Dgc_RevocationChunk>(
+			signatureVerifier: signatureVerifier
+		)
+	}
+	
+	// MARK: - Protocol Resource
+
+	typealias Send = EmptySendResource
+	typealias Receive = ProtobufReceiveResource<SAP_Internal_Dgc_RevocationChunk>
+
+	let trustEvaluation: TrustEvaluating
+	let locator: Locator
+	let type: ServiceType
+	let sendResource: EmptySendResource
+	let receiveResource: ProtobufReceiveResource<SAP_Internal_Dgc_RevocationChunk>
+	
+	func customError(for error: ServiceError<KIDTypeChunkResourceError>, responseBody: Data?) -> KIDTypeChunkResourceError? {
+		switch error {
+		case .resourceError:
+			return .DCC_RL_KTXY_INVALID_SIGNATURE
```

Comment:
Is it really the only error we have here?

Reply:
As far as I understand the tech spec right, there is only an custom error for failed signature validation. I have added a check for .signatureVerification error.